src/main/kotlin/io/snyk/plugin/ui/toolwindow/SnykToolWindowPanel.kt

```diff
@@ -506,6 +540,17 @@ class SnykToolWindowPanel(val project: Project) : JPanel(), Disposable {
         )
     }
 
+    private fun enableProductsAccordingToServerSetting() {
+        pluginSettings().apply {
+            sastOnServerEnabled = service<SnykApiService>().sastOnServerEnabled
```

Comment:
null case (when ApiClient fail to initioalize) should be handled. afaik we treat it as `false`